sky/shell/platform/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -23,163 +22,163 @@
-    flags_ = node->flags.Pass();
+  if (!mojoNode->flags.is_null()) {
+    _flags = mojoNode->flags.Pass();
   }
 
-  if (!node->strings.is_null()) {
-    strings_ = node->strings.Pass();
+  if (!mojoNode->strings.is_null()) {
+    _strings = mojoNode->strings.Pass();
   }
 
-  if (!node->geometry.is_null()) {
-    geometry_ = node->geometry.Pass();
+  if (!mojoNode->geometry.is_null()) {
+    _geometry = mojoNode->geometry;
   }
 
-  if (!node->children.is_null()) {
-    // Mark children for removal
-    for (scoped_refptr<Node> child : children_) {
-      DCHECK(child->parent_ != nullptr);
-      child->parent_ = nullptr;
+  if (!mojoNode->children.is_null()) {
+    // Mark existing children as orphans
+    NSArray* oldChildren = _children;
+    for (AccessibilityNode* child in oldChildren) {
+      DCHECK(child->_parent != nil);
+      child->_parent = nil;
     }
 
     // Set the new list of children
-    std::vector<scoped_refptr<Node>> children;
-    for (const semantics::SemanticsNodePtr& childNode : node->children) {
-      scoped_refptr<Node> child = bridge_->UpdateNode(childNode);
-      child->parent_ = this;
-      children.push_back(child);
+    NSMutableArray* children = [[NSMutableArray alloc] init];
+    _children = children;
+    for (const semantics::SemanticsNodePtr& mojoChild : mojoNode->children) {
+      AccessibilityNode* child = _bridge->UpdateNode(mojoChild);
+      child->_parent = self;
+      [children insertObject:child atIndex:0];
     }
-    children.swap(children_);
 
-    // Remove those children that are still marked for removal
-    for (scoped_refptr<Node> child : children) {
-      if (child->parent_ == nullptr) {
-        bridge_->RemoveNode(child);
+    // Remove those children that are still marked as orphans
+    for (AccessibilityNode* child in oldChildren) {
+      if (child->_parent == nil) {
+        _bridge->RemoveNode(child);
       }
     }
+    [oldChildren release];
   }
+}
 
-  global_transform_.release();
-  global_rect_.release();
+- (void)remove {
+  _parent = nil;
+  _bridge->RemoveNode(self);
 }
 
-void AccessibilityBridge::Node::ValidateGlobalTransform() {
-  if (global_transform_ != nullptr) {
-    return;
-  }
+#pragma mark - UIAccessibility overrides
 
-  if (parent_ == nullptr) {
-    global_transform_.reset(new SkMatrix44(geometry_.transform));
-  } else {
-    parent_->ValidateGlobalTransform();
-    global_transform_.reset(
-        new SkMatrix44(geometry_.transform * *(parent_->global_transform_)));
-  }
+- (BOOL)isAccessibilityElement {
+  // Per UIAccessibilityContainer, accessibility containers MUST return
+  // NO to -isAccessibilityElement
+  return (_children.count == 0);
 }
 
-void AccessibilityBridge::Node::ValidateGlobalRect() {
-  if (global_rect_ != nullptr) {
-    return;
-  }
+- (NSString*)accessibilityLabel {
+  return (_strings.is_null() || _strings->label.get().empty())
+             ? nil
+             : [NSString stringWithUTF8String:_strings->label.data()];
```

Comment:
You don't have to, but you could have also used `@(_string->label.data())` instead of the long form.